pkg/kubectl/cmd/templates/command_groups.go

```diff
@@ -59,3 +66,25 @@ func AddAdditionalCommands(g CommandGroups, message string, cmds []*cobra.Comman
 	}
 	return append(g, group)
 }
+
+func CheckRequiredFlags(cmd *cobra.Command, _ []string) error {
+	flags := cmd.Flags()
+	missingFlagNames := []string{}
+
+	flags.VisitAll(func(flag *pflag.Flag) {
+		requiredAnnotation, found := flag.Annotations[cobra.BashCompOneRequiredFlag]
```

Comment:
@liggitt For the annotation inversion,
> these annotations are for bash completion generation, not to govern behavior when running the command

Actually when a flag is marked as required, [`MarkFlagRequired`](https://github.com/spf13/cobra/blob/master/bash_completions.go#L504-L507) will call `flags.SetAnnotation`. These annotations are the only place to find all required flags. We just use the annotations data here.